app/ide-desktop/lib/dashboard/src/components/Portal/PortalProvider.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * @file
+ * Provides the context for the Portal component
+ */
+import * as React from 'react'
+
+import invariant from 'tiny-invariant'
+
+const PortalContext = React.createContext<React.RefObject<Element | null> | null>(null)
+
+/**
+ * Allows to access the root element for the Portal component
+ */
+export function usePortalContext() {
+  const root = React.useContext(PortalContext)
+
+  return { root } as const
+}
+
+/**
+ * Allows to access the root element for the Portal component
+ * @throws invariant the `PortalProvider` is not in the component tree
+ */
+export function useStrictPortalContext() {
+  const root = React.useContext(PortalContext)
+
+  invariant(
```

Comment:
invariant is a nice wrapper around.

Reply:
while that's fair, i'm not sure it's worth pulling in a dependency + making the code less obvious unless you already know what `invariant` does.

most importantly: it's not like it's even any faster to read, nor is it much shorter

Reply:
invariant is a standard thing in js ecosystem: https://www.npmjs.com/package/tiny-invariant. Also it matches typescript's `assert` pattern: https://blog.logrocket.com/assertion-functions-typescript/

Reply:
it's *popular*, yes. but that doesn't mean it's *standard*, nor does it mean we have to use it just because it exists.
see for example other libraries that we don't use:
<https://www.npmjs.com/package/lodash>
<https://www.npmjs.com/package/jquery>
<https://www.npmjs.com/package/@sinclair/typebox>

in general: i'm not sure about pulling in utility functions that don't really help make the code clearer